src/content/docs/game-design/godot/dungeoncrawler/3-playerimprovement/index.mdx

```diff
@@ -0,0 +1,199 @@
+
+2. Then, open the **Input Map** tab.
+
+3. We'll add our new Inputs by typing a name for each of them in the **Add New Action** box and hitting add. 
+
+	I'll be calling mine "Up" "Down" "Left" "Right" and i'll also add one called "Attack"
+
+	![Input Actions](/src/assets/godot/dungeonCrawler/inputmappreinputs.png)
+
+4. To add buttons to these, we'll hit the "+" to the right of each action, we'll then physically press the key we want to assign on our keyboard, and hit "Ok"
+
+	For my **Attack** action i'll be using **Left Click* for which you'll want to physically click *once* inside the **Listening for input** box.
+
+	Here's what my inputs look like:
+
+	![Input Actions with buttons](/src/assets/godot/dungeonCrawler/inputswithbuttons.png)
+
+	If you're happy with your inputs, you can then hit **Close** at the bottom of the screen. Let's navigate back to our **player_movement.gd** script. This can be done by opening our **player** scene, clicking on the **CharacterBody2D** and clicking the **Script** tab at the top of the screen.
+
+5. To change our inputs, all we need to do is change the predefined inputs to what we called ours!
+
+	:::Note[Case Sensitivity] 
+	What you named your input actions *is* case sensitive
+	:::
+
+	So, "ui_left" becomes "Left"
+	"ui_right" becomes "Right" and so on!
+
+	leaving that line in our script looking like this:
+
+	```gdscript
+    	var direction = Input.get_vector("Left", "Right", "Up", "Down")
+	```
+</Steps>
+if you run your game, you'll notice your inputs are now changed!
+
+We won't assign our Attack option just yet, but it's good we've created it already.
+
+
+### Animation switching
+
+To change our animation depending on if we're moving or not, we'll only need to add a few lines to our script. But first, we'll need a reference to our **AnimatedSprite2D** in our script.
+
+We can do this easily, by **Clicking** and **Dragging** our **AnimatedSprite2D** into our script, making sure we hold **Ctrl** on the keyboard after grabbing it, but before dropping it.
+
+We'll want to drop it below the speed variable declaration
+
+```gdscript
+@export var speed = 200
+@onready var animated_sprite_2d = $AnimatedSprite2D
+```
+
+if it doesn't look like this, delete the line and try again, making sure you're holding **Ctrl** on your keyboard after picking it up.
+
+Great! Let's get to changing our animation!
+
+<Steps>
+
+1. Under where we set our velocity, we'll simply add a check to see if our velocity is anything other than 0 and change our animation based on that!
+
+	We can check this simply with:
+
+	```gdscript
+	if velocity != Vector2.ZERO:
+	```
+	:::note[Vector2s] 
+	Why can't we just do velocity != 0? This is because velocity actually contains both our X (Horizontal) and Y (Vertical) velocity, so we need to make sure **Both** aren't zero.
+	:::
+
+2. Then, we can play our run animation!
+
+	```gdscript
+	if velocity != Vector2.ZERO:
```

Comment:
Please make sure indentation for all parts of the code is correct so can be copy/paster if wanted